modules/zivid/presets.py

```diff
@@ -0,0 +1,144 @@
+
+    def __str__(self):
+        return str(self.__impl)
+
+
+class Category:
+    """Preset category.
+
+    This class cannot be initialized directly by the end-user. Use the categories function to
+    obtain a list of categories available for a camera.
+    """
+
+    def __init__(self, impl):
+        """Initialize Category wrapper.
+
+        This constructor is only used internally, and should not be called by the end-user.
+
+        Args:
+            impl:   Reference to internal/back-end instance.
+
+        Raises:
+            TypeError: If argument does not match the expected internal class.
+        """
+        if not isinstance(impl, _zivid.presets.Category):
+            raise TypeError(
+                "Unsupported type for argument impl. Got {}, expected {}".format(
+                    type(impl), type(_zivid.presets.Category)
+                )
+            )
+        self.__impl = impl
+
+    @property
+    def name(self):
+        """Get the name of the category.
+
+        Returns:
+            The name of the category.
+        """
+        return self.__impl.name()
+
+    @property
+    def presets(self):
+        """Get the presets available in the category.
+
+        The settings might change between different releases of the SDK. New presets might be added
+        and old ones might be removed. If having the exact same settings is desired, it is
+        recommended to save them to a YML file and load as needed.
+
+        Returns:
+            The presets available in the category.
+        """
+        return [Preset(p) for p in self.__impl.presets()]
+
+    def __str__(self):
+        return str(self.__impl)
+
+
+def categories(model):
+    """Get available preset categories for the specified camera model.
+
+    A preset category contains a collection of presets optimized for one scenario or use case.
+    scenarios.
+
+    The settings might change between different releases of the SDK. New presets might be added
+    and old ones might be removed. If having the exact same settings is desired, it is recommended
+    to save them to a YML file and load as needed.
+
+    Args:
+        model: The model for the camera whose preset categories should be returned. This value can
+               be obtained from CameraInfo.model.
+
+    Raises:
+        ValueError: if model is invalid.
+
+    Returns:
+        The available categories for the specified camera model.
+    """
+    if model not in CameraInfo.Model.valid_values():
+        raise ValueError(
+            "Invalid camera model. Got {}, expected one of {}".format(
+                model, CameraInfo.Model.valid_values()
+            )
+        )
+    # categories wants a _zivid.CameraInfo.Model. We need an _zivid.CameraInfo.Model.enum to
+    # construct it.
+    _model = _zivid.CameraInfo.Model(getattr(_zivid.CameraInfo.Model, model))
+    return [Category(c) for c in _zivid.presets.categories(_model)]
```

Comment:
I don't like using `getattr` here as it relies on names of the fields being the same as their values. How about this:
```suggestion
    return [
        Category(c) for c in _zivid.presets.categories(_to_internal_camera_info(CameraInfo(model=model)).model)
    ]
```